internal/jujuclient/dial.go

```diff
@@ -362,6 +362,25 @@ func (c *Connection) ConnectStream(path string, attrs url.Values) (base.Stream,
 // values are used as URL query values when making the initial
 // HTTP request. Headers passed in will be added to the HTTP
 // request.
-func (c *Connection) ConnectControllerStream(path string, attrs url.Values, headers http.Header) (base.Stream, error) {
-	return nil, errors.E(errors.CodeNotImplemented)
+func (c *Connection) ConnectControllerStream(path string, attrs url.Values, extraHeaders http.Header) (base.Stream, error) {
+	const op = errors.Op("jujuclient.ConnectControllerStream")
+
+	user, pass, err := c.dialer.ControllerCredentialsStore.GetControllerCredentials(c.ctx, c.ctl.Name)
```

Comment:
Because we're getting controller credentials here, and it's sensitive, this might be a candidate for an auth check? (If it's hassle don't bother)